pixelgl/canvas.go

```diff
@@ -37,32 +37,37 @@ func NewCanvas(bounds pixel.Rect) *Canvas {
 		col: mgl32.Vec4{1, 1, 1, 1},
 	}
 
+	baseShader(c)
 	c.SetBounds(bounds)
+	c.shader.update()
+	return c
+}
 
-	var shader *glhf.Shader
-	mainthread.Call(func() {
-		var err error
-		shader, err = glhf.NewShader(
-			canvasVertexFormat,
-			canvasUniformFormat,
-			canvasVertexShader,
-			canvasFragmentShader,
-		)
-		if err != nil {
-			panic(errors.Wrap(err, "failed to create Canvas, there's a bug in the shader"))
-		}
-	})
-	c.shader = shader
+// BindUniform will add a uniform with any supported underlying variable
+// if the uniform already exists, including defaults, they will be reassigned
+// to the new value
+func (c *Canvas) BindUniform(Name string, Value interface{}) {
+	c.shader.AddUniform(Name, Value)
+}
 
-	return c
+// UpdateShader needs to be called after any changes to the underlying GLShader
+// are made (ie, BindUniform(), SetFragmentShader()...)
+func (c *Canvas) UpdateShader() {
+	c.shader.update()
+}
+
+// SetFragmentShader allows you to define a new fragment shader on the underlying
```

Comment:
not sure define is the right word... I guess 'set' (just like in the method name) is better